File: unit/src/main/java/net/mekstrike/unit/Battlefield.java
package net.mekstrike.unit;

import io.dapr.actors.ActorMethod;
import io.dapr.actors.ActorType;

@ActorType(name = "battlefield")
public interface Battlefield {
    @ActorMethod(name = "GetNumberOfCols")
    public int getNumberOfCols();

    @ActorMethod(name = "GetNumberOfRows")
    public int getNumberOfRows();

    @ActorMethod(name = "IsCellBlocked")
    public boolean isCellBlocked(CellRef cellRef);
    
    @ActorMethod(name = "BlockCell")
    public void blockCell(CellRef cellRef);
}

File: battlefield/Battlefield.Interfaces/Battlefield.cs
using Dapr.Actors;
using System.Threading.Tasks;
using System.Collections.Generic;

namespace Battlefield.Interfaces
{
    public interface IBattlefield : IActor
    {
        Task<List<Hex>> GetBoardCells();

        Task<List<Hex>> GetMovementOptions(int startX, int startY, string movement);

        Task<int>GetNumberOfCols();

        Task<int>GetNumberOfRows();

        Task<bool> IsCellBlocked(Position position);
        Task BlockCell(Position position);
    }
}